Add isset() check to prevent logging warnings.

Sageth opened the pull request:
In some cases, viewing a page or post will cause a PHP warning:
```
PHP Warning: Trying to access array offset on value of type null in /wp-includes/widgets.php on line 918" while reading response header from upstream, ...,
request: "GET /attachment-sitemap4.xml HTTP/2.0", ...,
referrer: "/sitemap_index.xml"
```

This appears to be a logging issue only.

Description of Change: Adding a check to see if the widget ID is set prior to evaluation. 

Trac ticket: https://core.trac.wordpress.org/ticket/57518

Sageth committed:
Add isset() check to prevent logging warnings.

In some cases, viewing a file will cause a PHP warning:
```
PHP Warning: Trying to access array offset on value of type null in /wp-includes/widgets.php on line 918" while reading response header from upstream, ...,
request: "GET /attachment-sitemap4.xml HTTP/2.0", ...,
referrer: "/sitemap_index.xml"
```

This appears to be a logging issue only.

mindctrl commented on src/wp-includes/widgets.php:
```diff
@@ -915,7 +915,7 @@ function is_active_widget( $callback = false, $widget_id = false, $id_base = fal
 				foreach ( $widgets as $widget ) {
 					if ( ( $callback && isset( $wp_registered_widgets[ $widget ]['callback'] ) && $wp_registered_widgets[ $widget ]['callback'] === $callback ) || ( $id_base && _get_widget_id_base( $widget ) === $id_base ) ) {
-						if ( ! $widget_id || $widget_id === $wp_registered_widgets[ $widget ]['id'] ) {
+						if ( ! $widget_id || isset( $wp_registered_widgets[ $widget ]['id'] ) && $widget_id === $wp_registered_widgets[ $widget ]['id'] ) {
```
My IDE suggests wrapping the latter conditions in parenthesis because operations priority might differ from what's expected. Like this:

```suggestion
						if ( ! $widget_id || ( isset( $wp_registered_widgets[ $widget ]['id'] ) && $widget_id === $wp_registered_widgets[$widget]['id'] ) ) {
```

mindctrl commented:
@Sageth are you still around and able to bring this up to date with `trunk`?

Sageth committed:
Merge branch 'WordPress:trunk' into logging

Sageth committed:
Add parenthesis to ensure operational priority

Co-authored-by: John Parris <public@johnparris.com>

Sageth committed:
Add spaces to confirm to standards

For resolving 

FILE: src/wp-includes/widgets.php
--------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
--------------------------------------------------------------------------------
 928 | ERROR | [x] Array keys must be surrounded by spaces unless they contain
     |       |     a string or an integer.
     |       |     (WordPress.Arrays.ArrayKeySpacingRestrictions.NoSpacesAroundArrayKeys)
--------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
--------------------------------------------------------------------------------

Sageth commented:
> @Sageth are you still around and able to bring this up to date with `trunk`?

Updated with `trunk` as well as adding parenthesis suggestion and addressing the code standard failure.
